File: backend/src/main/java/com/foodya/backend/application/dto/CreateOrderRequest.java
package com.foodya.backend.application.dto;

import java.math.BigDecimal;
import java.util.List;

public record CreateOrderRequest(
        String restaurantId,
        List<CreateOrderItemRequest> items,
        String deliveryAddress,
        BigDecimal deliveryLatitude,
        BigDecimal deliveryLongitude,
        String customerNote
) {
}

File: mobile/lib/features/customer/domain/models/create_order_request.dart
class CreateOrderRequest {
  const CreateOrderRequest({
    required this.deliveryAddress,
    required this.deliveryLatitude,
    required this.deliveryLongitude,
    this.customerNote,
  });

  final String deliveryAddress;
  final double deliveryLatitude;
  final double deliveryLongitude;
  final String? customerNote;

  Map<String, dynamic> toJson() {
    return {
      'deliveryAddress': deliveryAddress,
      'deliveryLatitude': deliveryLatitude,
      'deliveryLongitude': deliveryLongitude,
      if (customerNote != null && customerNote!.trim().isNotEmpty)
        'customerNote': customerNote!.trim(),
    };
  }
}
